### DISCUSSION TITLE
$persist in re-usable components

### DISCUSSION THREAD

**Discussion by su8898** [alpinejs/alpine#1932] [1. Help]
How do we register a variable as a `$persist` value inside re-usable components? 

For example, the following code will return an error that says `$persist is not defined`. The `'this` (as in `this.$refs` etc) keyword will not work on _Top-level_ either. 

```document.addEventListener('alpine:init', () => {
    Alpine.data('dropdown', () => ({
        open: $persist(false),

        toggle() {
            this.open = !this.open
        },
    }))
})```

**Comment by SimoTod** [comment]
If you want to use this at the top level, you need to use a standard function.
Alpine needs to bind a custom this to make magics available at the top level and in javascript is not possible to re-bind this for arrow functions.

```javascript
Alpine.data('dropdown', function() {
    return {
        open: $persist(false),

        toggle() {
            this.open = !this.open
        },
    }
})
```
